### PR TITLE
feat(core): support password change

### PR DESCRIPTION
**Pull request by maciez**

### CODE REVIEW

File: src/main/resources/static/apps/admin/modules/core/userProfile.html [hunk 1]
```diff
@@ -52,6 +53,35 @@
                         </div>
                     </div>
                 </form>
+
+                <form name="pwdForm" class="col-sm-8 col-sm-offset-2">
+                    <h2 class="with-border">{{'admin.modules.core.userProfile.titleChangePassword'|translate}}</h2>
+                    <div class="row">
+                        <div class="col-xs-7">
+                            <div class="form-group" ng-class="{'has-error': pwdForm.oldPassword.$dirty && pwdForm.oldPassword.$touched && pwdForm.oldPassword.$invalid}">
```
Comment (bndynet) [inline src/main/resources/static/apps/admin/modules/core/userProfile.html]: I wrote a component for form-group, you can use it instead. For example:
```html
<ui-input name="realName"
                label="admin.modules.core.userProfile.labelRealName"
                ng-model="viewModel.realName"></ui-input>
```

File: src/main/java/net/bndy/wf/modules/core/api/UserController.java [hunk 2]
```diff
@@ -129,6 +130,14 @@ public File upload(MultipartFile file, HttpServletRequest request) throws Illega
         return f;
     }
 
+    @RequestMapping(value = "/changePassword", method = RequestMethod.POST)
+    public void changePassword(@RequestBody Map<String, String> data) {
+        User u = this.getCurrentUser();
+        if (this.userService.comparePassword(data.get("oldPassword"), u.getPassword())) {
+            this.userService.changePassword(u.getId(), data.get("newPassword"));
+        }
```
Comment (bndynet) [inline src/main/java/net/bndy/wf/modules/core/api/UserController.java]:
```java
else {
}
```
If password is error, should throw an exception.

File: src/main/resources/static/apps/admin/modules/core/userProfile.html (hunk 1 above)
Comment (maciez) [reply src/main/resources/static/apps/admin/modules/core/userProfile.html]: :clap: Cool!

File: src/main/java/net/bndy/wf/modules/core/api/UserController.java (hunk 2 above)
Comment (maciez) [reply src/main/java/net/bndy/wf/modules/core/api/UserController.java]: I will implement it later.

**Comment by bndynet** [comment]
closed #36

**Pull request merged**